modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java

```diff
@@ -0,0 +1,424 @@
+								false);
+
+							LayoutSEOLink layoutSEOLink =
+								_layoutSEOLinkManager.getCanonicalLayoutSEOLink(
+									layout, locale, canonicalURL,
+									_portal.getAlternateURLs(
+										canonicalURL, themeDisplay, layout));
+
+							return layoutSEOLink.getHref();
+						}
+						catch (PortalException portalException) {
+							_log.error(portalException, portalException);
+
+							return StringPool.BLANK;
+						}
+						finally {
+							httpServletRequest.setAttribute(
+								LayoutDisplayPageWebKeys.
+									LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER,
+								initialLayoutDisplayPageObjectProvider);
+						}
+					}
+				).orElse(
+					StringPool.BLANK
+				);
+			}
+		).orElse(
+			StringPool.BLANK
+		);
+	}
+
+	@Override
+	public Locale getDefaultLocale(
+		LayoutDisplayPageObjectProvider layoutDisplayPageObjectProvider) {
+
+		return Optional.ofNullable(
+			_groupLocalService.fetchGroup(
+				layoutDisplayPageObjectProvider.getGroupId())
+		).map(
+			Group::getDefaultLanguageId
+		).map(
+			LocaleUtil::fromLanguageId
+		).orElseGet(
+			LocaleUtil::getDefault
+		);
+	}
+
+	@Override
+	public Date getPublishDate(
+		LayoutDisplayPageObjectProvider layoutDisplayPageObjectProvider) {
+
+		try {
+			ClassName className = _classNameLocalService.getClassName(
+				layoutDisplayPageObjectProvider.getClassNameId());
+
+			InfoItemFieldValuesProvider infoItemFieldValuesProvider =
+				_infoItemServiceTracker.getFirstInfoItemService(
+					InfoItemFieldValuesProvider.class,
+					className.getClassName());
+
+			return Optional.ofNullable(
+				infoItemFieldValuesProvider.getInfoItemFieldValue(
+					layoutDisplayPageObjectProvider.getDisplayObject(), "publishDate")
+			).filter(
+				infoFieldValue -> {
+					InfoField infoField = infoFieldValue.getInfoField();
+
+					return Objects.equals(
+						infoField.getInfoFieldType(),
+						DateInfoFieldType.INSTANCE);
+				}
+			).map(
+				infoFieldValue -> (Date)infoFieldValue.getValue()
+			).orElseGet(
+				() -> new Date()
+			);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException, portalException);
+		}
+
+		return new Date();
+	}
+
+	@Override
+	public String getTitle(
+		LayoutDisplayPageObjectProvider layoutDisplayPageObjectProvider,
+		Locale locale) {
+
+		try {
+			ClassName className = _classNameLocalService.getClassName(
+				layoutDisplayPageObjectProvider.getClassNameId());
+
+			InfoItemFieldValuesProvider infoItemFieldValuesProvider =
+				_infoItemServiceTracker.getFirstInfoItemService(
+					InfoItemFieldValuesProvider.class,
+					className.getClassName());
+
+			return Optional.ofNullable(
+				infoItemFieldValuesProvider.getInfoItemFieldValue(
+					layoutDisplayPageObjectProvider.getDisplayObject(), "title")
+			).filter(
+				infoFieldValue -> {
+					InfoField infoField = infoFieldValue.getInfoField();
+
+					return Objects.equals(
+						infoField.getInfoFieldType(),
+						DateInfoFieldType.INSTANCE);
```

Comment:
Should "title" use a `DateInfoFieldType` type?

Reply:
I would say that NO